code/online_challenges/src/hackerrank/Sock_Merchant/Sock_Merchant.cpp

```diff
@@ -0,0 +1,31 @@
+#include <iostream>
+#include <bits/stdc++.h>
+using namespace std;
+
+
+
+int main()
+{
+    int n;
+    cin >> n;
+    int arr[n];
```

Comment:
Use std::vector (unless this is hackerrank template). Use pre-increment in alll loops.